lib/thrift/transport/ssl.ex

```diff
@@ -30,6 +31,11 @@ defmodule Thrift.Transport.SSL do
     end
   end
 
+  @spec optional?([option]) :: {boolean, [:ssl.ssloption]} | {:error, Exception.t}
+  def optional?(opts) do
+    Keyword.pop(opts, :optional, false)
+  end
```

Comment:
I think it would simplify the API if we merged `configuration` and `optional` so that `configuration` return `{:require | :optional, opts} | ..}`.

Reply:
That's a great idea, I'll do that